backend_test.go

```diff
@@ -133,6 +136,23 @@ func testBackendGetSetDelete(t *testing.T, backend BackendType) {
 	value, err = db.Get([]byte("x"))
 	require.NoError(t, err)
 	require.Equal(t, []byte{}, value)
+
+	err = db.Compact(nil, nil)
+	if strings.Contains(string(backend), "pebbledb") {
+		// In pebble the start and end will be the same so
+		// we expect an error
+		require.Error(t, err)
+	}
+
+	err = db.Set([]byte("y"), []byte{})
+	require.NoError(t, err)
+
+	err = db.Compact(nil, nil)
+	require.NoError(t, err)
+
+	if strings.Contains(string(backend), "pebbledb") {
```

Comment:
please add a comment re

"When running the test the folder can't be cleaned up and there is a panic on removing the tmp testing directories.. The compcation process is slow to release the lock on the folder."